src/main/java/com/openlattice/datastore/services/EdmService.java

```diff
@@ -469,18 +474,20 @@ public void createEntitySet( Principal principal, EntitySet entitySet, Set<UUID>
                     principal,
                     EnumSet.allOf( Permission.class ) );
 
-            ownablePropertyTypes.stream()
+            ownablePropertyTypeIDs.stream()
                     .map( propertyTypeId -> new AclKey( entitySet.getId(), propertyTypeId ) )
                     .peek( aclKey -> {
                         authorizations.setSecurableObjectType( aclKey,
                                 SecurableObjectType.PropertyTypeInEntitySet );
-                    } ).forEach( aclKey -> authorizations.addPermission(
-                    aclKey,
-                    principal,
-                    EnumSet.allOf( Permission.class ) ) );
+                    } )
+                    .forEach( aclKey -> authorizations.addPermission( aclKey, principal, EnumSet.allOf( Permission.class ) ) );
+
+            List<PropertyType> ownablePropertyTypes = Lists.newArrayList( propertyTypes.getAll( ownablePropertyTypeIDs ).values() );
```

Comment:
Is there a reason you need to copy from Collection into List?

Reply:
In the ElasticSearchApi the saveEntitySetToElasticsearch is declared with a List as input, which is called from EsEdmService